src/java/org/apache/cassandra/utils/memory/MemtableCleanerThread.java

```diff
@@ -27,54 +33,82 @@
  */
 public class MemtableCleanerThread<P extends MemtablePool> extends InfiniteLoopExecutor
 {
-    private static class Clean<P extends MemtablePool> implements InterruptibleRunnable
+    private static final Logger logger = LoggerFactory.getLogger(MemtableCleanerThread.class);
+    public static class Clean<P extends MemtablePool> implements InterruptibleRunnable
     {
+        /** This is incremented when a cleaner is invoked and decremented when a cleaner has completed */
+        final AtomicInteger numPendingTasks = new AtomicInteger(0);
+
         /** The pool we're cleaning */
         final P pool;
 
         /** should ensure that at least some memory has been marked reclaiming after completion */
-        final Runnable cleaner;
+        final MemtableCleaner cleaner;
 
         /** signalled whenever needsCleaning() may return true */
         final WaitQueue wait = new WaitQueue();
 
-        private Clean(P pool, Runnable cleaner)
+        private Clean(P pool, MemtableCleaner cleaner)
         {
             this.pool = pool;
             this.cleaner = cleaner;
         }
 
-        boolean needsCleaning()
+        /** Return the number of pending tasks */
+        public int numPendingTasks()
         {
-            return pool.onHeap.needsCleaning() || pool.offHeap.needsCleaning();
+            return this.numPendingTasks.get();
```

Comment:
`this` is not required:
```suggestion
            return numPendingTasks.get();
```